ai/integrations/litellm/src/unitycatalog/ai/litellm/utils.py

```diff
@@ -0,0 +1,172 @@
+        return str(result.value)
+
+    def to_tool_result_message(self, result: str) -> dict[str, Any]:
+        """
+        Creates a tool result message based on the result of the function execution.
+
+        Args:
+            result (str): The result of the tool execution.
+
+        Returns:
+            dict[str, Any]: A dictionary representing the tool result message.
+        """
+        return {
+            "role": "tool",
+            "tool_call_id": self.tool_call_id,
+            "name": self.function_name,
+            "content": result,
+        }
+
+
+def _extract_tool_call_data_from_choice(choice: Choices) -> list[ToolCallData]:
+    tool_calls_data = []
+
+    if choice.finish_reason == "tool_calls":
+        if choice.message and (tool_calls := choice.message.tool_calls):
+            for tool_call in tool_calls:
+                if isinstance(tool_call, ChatCompletionMessageToolCall):
+                    tool_calls_data.append(
+                        ToolCallData(
+                            function_name=construct_original_function_name(tool_call.function.name),
+                            arguments=json.loads(tool_call.function.arguments),
+                            tool_call_id=tool_call.id,
+                        )
+                    )
+
+    return tool_calls_data
+
+
+def extract_tool_call_data(response: Message) -> list[list[ToolCallData]]:
+    """
+    Extracts the tool call information from the LiteLLM response. Note there can be multiple tool
+    choices for a single LiteLLM Message, and for each choice there can be multiple tool calls.
+
+    Args:
+        response (Message): The response from LiteLLM optionally containing tool usage blocks.
+
+    Returns:
+        list[list[ToolCallData]]: A list of ToolCallData objects containing function names,
+        arguments, and tool use IDs.
+    """
+    return [_extract_tool_call_data_from_choice(choice) for choice in response.choices or []]
+
+
+def generate_tool_call_messages(
+    *,
+    response: Message,
+    conversation_history: Union[dict[str, Any], list[dict[str, Any]]],
+    client: Optional[BaseFunctionClient] = None,
+    choice_index: int = 0,
+) -> list[dict[str, Any]]:
+    """
+    Generate tool call messages from the response.
+
+    If there are multiple tool calls in the selected Choice, each tool's function call response will
+    be appended to the messages as a unique message. For instance, if there are 2 tool calls
+    requested by the LiteLLM response, we will append the following payload to the
+    `conversation_history` and `response` list.
+
+    [
+        {"role", "user": "content": "function_1_response_as_a_string"},
+        {"role", "user": "content": "function_2_response_as_a_string"}
+    ]
+
+    Note:
+        This function relies on that the UC function names don't contain '__' in the catalog, schema
+        or function names, and the total length of the function name is less than 64 characters.
```

Comment:
Where does the 64 character limit come from? 
Do we need to enforce this in the toolkit?

Reply:
Could we add a test to validate the behavior after adding some sort of correctness guard?

Reply:
This came from the OpenAI tool spec which we were adhering to, given LiteLLM does not have a standard spec and typically leverages OpenAI's. However, on second thought that limitation is not necessary. Great catch thanks